src/metabase/types.clj

```diff
@@ -5,6 +5,7 @@
 (derive :type/Dictionary :type/Collection)
 (derive :type/Array :type/Collection)
 
+(derive :type/Nil :type/*)
```

Comment:
I don't think the added complexity makes sense especially since it doesn't look like you actually do anything with `:type/Nil`. Don't forget you can just use `s/maybe` for something that might be `nil`. That's what I think you should do here if you're using this for a schema